x-pack/legacy/plugins/alerting/server/init.ts

```diff
@@ -77,15 +117,37 @@ export function init(server: Legacy.Server) {
   server.decorate('request', 'getAlertsClient', function() {
     const request = this;
     const savedObjectsClient = request.getSavedObjectsClient();
+
     const alertsClient = new AlertsClient({
       log: server.log.bind(server),
       savedObjectsClient,
       alertTypeRegistry,
       taskManager: taskManager!,
       spaceId: spaces.isEnabled ? spaces.getSpaceId(request) : undefined,
+      async getUserName(): Promise<string | null> {
+        if (!security.isEnabled) {
+          return null;
+        }
+        const user = await security.authc.getCurrentUser(KibanaRequest.from(request));
+        return user ? user.username : null;
+      },
+      async createAPIKey(): Promise<CreateAPIKeyResult> {
+        if (!security.isEnabled) {
+          return { created: false };
+        }
+        return {
+          created: true,
+          result: (await security.authc.createAPIKey(KibanaRequest.from(request), {
+            name: Date.now().toString(),
```

Comment:
I am not sure what we should put here for a name. We don't have an alert id just yet because we need the API key to create an alert. It seems to require something unique, maybe uuid?

Reply:
Fixed in https://github.com/elastic/kibana/pull/41389/commits/bdef9ed34ee5ea2a0b820435e2db8be7205c2037.